src/main/java/uk/ac/sanger/sccp/stan/service/FindService.java

```diff
@@ -259,6 +296,27 @@ public List<StoredItem> getStoredItems(List<LabwareSample> labwareSamples) {
         return storeService.getStored(labwareBarcodes);
     }
 
+    /**
+     * Fetches works numbers for the given labware and sample
+     * @param labware the labware to search for
+     * @param sample the sample to search for
+     * @return a set of work numbers
+     */
+    public Set<String> getWorkNumbers(Labware labware, Sample sample) {
+        if (labware == null || sample == null) {
+            return Set.of();
+        }
+        return labware.getSlots().stream().flatMap(slot -> {
+            if (slot.getSamples() != null && !slot.getSamples().isEmpty() && slot.getSamples().contains(sample)) {
```

Comment:
You don't need `!slot.getSamples().isEmpty()` if you're about to do `slot.getSamples().contains(...)`